### DISCUSSION TITLE
export trajectory trace

### DISCUSSION THREAD

**Discussion by sampson-young** [HALRobotics/Beta#105] [Q&A]
Hi HAL,

I am using HAL for a robotic trajectory simulation. 
Is there any method to **export**:
1. the trajectory traces into curves (The blue trajectory showed in the pic below)?
2. the target planes?
3. the tool TCP plane with the axis (the red/gree/blue showed in the pic below)? 

![image](https://user-images.githubusercontent.com/37079727/122652446-cea08e80-d136-11eb-863b-ffe04761e1d1.png)

**Comment by thibaultschwartz** [comment]
Hi Sampson,
We added some monitoring features in the upcoming version so users can get the current joint values, tool position, robot meshes etc.
We did not plan to output the trace though, can you please explain how you would use it?
